hera_cal/reflections.py

```diff
@@ -267,25 +228,439 @@ def write_reflections(self, output_calfits, input_calfits=None, overwrite=False,
             if cal.Ntimes > self.Ntimes:
                 time_array = cal.time_array
             else:
-                time_array = self.reflection_times
+                time_array = self.times
             if cal.Nfreqs > self.Nfreqs:
                 freq_array = cal.freq_array
             else:
                 freq_array = self.freqs
             kwargs = dict([(k, getattr(cal, k)) for k in ['gain_convention', 'x_orientation',
                                                           'telescope_name', 'cal_style']])
         else:
-            time_array = self.reflection_times
+            time_array = self.times
             freq_array = self.freqs
             kwargs = {}
 
-        uvc = io.write_cal(output_calfits, rgains, freq_array, time_array, flags=flags, quality=quals, 
-                           total_qual=tquals, outdir=os.path.dirname(output_calfits), zero_check=False, 
-                           overwrite=overwrite, history=version.history_string(add_to_history), **kwargs)
+        echo("...writing {}".format(output_calfits))
+        uvc = io.write_cal(output_calfits, rgains, freq_array, time_array, flags=flags,
+                           quality=quals, total_qual=tquals, zero_check=False,
+                           overwrite=overwrite, history=version.history_string(add_to_history),
+                           **kwargs)
+
+        if write_npz:
+            output_npz = os.path.splitext(output_calfits)[0] + '.npz'
+            if not os.path.exists(output_npz) or overwrite:
+                echo("...writing {}".format(output_npz))
+                np.savez(output_npz, delay=self.ref_dly, phase=self.ref_phs, amp=self.ref_amp,
+                         significance=self.ref_significance, times=self.times, freqs=self.freqs,
+                         lsts=self.lsts, antpos=self.antpos,
+                         history=version.history_string(add_to_history))
+
         return uvc
 
+    def pca_decomp(self, dly_range, dfft=None, flags=None, side='both', keys=None, overwrite=False, verbose=True):
+        """
+        Create a PCA-based model of the FFT data in dfft.
```

Comment:
More detailed info needed here. What kind of PCA? What are the vectors? Give some context for what dly_range means.